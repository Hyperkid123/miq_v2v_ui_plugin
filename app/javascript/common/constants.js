export const MAX_LENGTH_NAMES = 24;
export const MAX_LENGTH_DESCRIPTIONS = 128;

const maxLengthIs = __('Maximum length is %s characters.');
const requiredMaxLengthIs = __('Required. Maximum length is %s characters.');
const youHaveReachedMax = __('You have reached the maximum length of %s characters.');

export const validation = {
  name: {
    help: sprintf(maxLengthIs, MAX_LENGTH_NAMES),
    requiredMessage: sprintf(requiredMaxLengthIs, MAX_LENGTH_NAMES),
    maxLength: MAX_LENGTH_NAMES,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_NAMES)
  },
  description: {
    help: sprintf(maxLengthIs, MAX_LENGTH_DESCRIPTIONS),
    maxLength: MAX_LENGTH_DESCRIPTIONS,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_DESCRIPTIONS)
  }
};

export const OPENSTACK = 'openstack';
export const RHV = 'rhevm';

export const V2V_TARGET_PROVIDERS = [
  {
    name: __('Red Hat Virtualization'),
    id: RHV,
    type: 'ManageIQ::Providers::Redhat::InfraManager'
  },
  {
    name: __('Red Hat OpenStack Platform'),
    id: OPENSTACK,
    type: 'ManageIQ::Providers::Openstack::CloudManager'
  }
];

export const FETCH_TARGET_COMPUTE_URLS = {
  [RHV]:
    '/api/clusters?expand=resources' +
    '&attributes=ext_management_system.emstype,v_parent_datacenter,ext_management_system.name,hosts' +
    '&filter[]=ext_management_system.emstype=rhevm',
  [OPENSTACK]: '/api/cloud_tenants?expand=resources&attributes=ext_management_system.name,ext_management_system.id'
};
