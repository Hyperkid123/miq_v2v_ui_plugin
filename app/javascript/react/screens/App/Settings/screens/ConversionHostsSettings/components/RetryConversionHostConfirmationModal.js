import React from 'react';
import PropTypes from 'prop-types';
import { Button, Modal, Icon } from 'patternfly-react';

const RetryConversionHostConfirmationModal = ({
  show,
  conversionHostTaskToRetry,
  isPostingConversionHosts,
  hideConversionHostRetryModalAction,
  postConversionHostsAction,
  postConversionHostsUrl
}) => (
  <Modal show={show} onHide={hideConversionHostRetryModalAction} backdrop="static">
    <Modal.Header>
      <Modal.CloseButton onClick={hideConversionHostRetryModalAction} />
      <Modal.Title>{__('Retry Conversion Host Configuration')}</Modal.Title>
    </Modal.Header>
    <Modal.Body className="warning-modal-body">
      <div className="warning-modal-body--icon">
        <Icon type="pf" name="info" />
      </div>
      <div className="warning-modal-body--list">
        <h4>
          {__(
            'To retry configuration of the conversion host below, re-enter the required SSH keys for Authentication.'
          )}
        </h4>
        <div>
          <ul>
            <h4>
              <strong>{conversionHostTaskToRetry && conversionHostTaskToRetry.name}</strong>
              {/* TODO also include the two SSH key fields. redux-form? maybe just local state? */}
            </h4>
          </ul>
        </div>
      </div>
    </Modal.Body>
    <Modal.Footer>
      <Button bsStyle="default" className="btn-cancel" onClick={hideConversionHostRetryModalAction}>
        {__('Cancel')}
      </Button>
      <Button
        bsStyle="primary"
        disabled={isPostingConversionHosts} // TODO disable if the ssh key fields are empty, too
        onClick={() => {
          const postBody = { ...conversionHostTaskToRetry.context_data.request_params };
          // TODO also pass the two SSH keys
          postConversionHostsAction(postConversionHostsUrl, [postBody]);
        }}
      >
        {__('Retry')}
      </Button>
    </Modal.Footer>
  </Modal>
);

RetryConversionHostConfirmationModal.propTypes = {
  show: PropTypes.bool,
  conversionHostTaskToRetry: PropTypes.object,
  isPostingConversionHosts: PropTypes.bool,
  hideConversionHostRetryModalAction: PropTypes.func,
  postConversionHostsAction: PropTypes.func,
  postConversionHostsUrl: PropTypes.string
};

export default RetryConversionHostConfirmationModal;
