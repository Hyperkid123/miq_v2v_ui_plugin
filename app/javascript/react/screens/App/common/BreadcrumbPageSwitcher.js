import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { OverlayTrigger, Popover, Button, Icon, ListGroup, ListGroupItem } from 'patternfly-react';

const BreadcrumbPageSwitcher = ({ items, activeHref }) => {
  const popover = (
    <Popover id="breadcrumb-switcher-popover">
      <ListGroup style={{ marginBottom: 0, borderTop: 0 }}>
        {items.map(({ name, href }) => (
          <ListGroupItem className={classNames('no-border', { active: href === activeHref })} href={href} key={href}>
            {name}
          </ListGroupItem>
        ))}
      </ListGroup>
    </Popover>
  );

  return (
    <OverlayTrigger rootClose trigger="click" placement="bottom" overlay={popover}>
      <Button
        style={{
          marginTop: '-2px',
          marginBottom: '-4px',
          marginLeft: '15px',
          height: '22px',
          width: '32px',
          lineHeight: '14px'
        }}
      >
        <Icon type="fa" name="exchange" />
      </Button>
    </OverlayTrigger>
  );
};

BreadcrumbPageSwitcher.propTypes = {
  activeHref: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string, href: PropTypes.string })).isRequired
};

export default BreadcrumbPageSwitcher;
