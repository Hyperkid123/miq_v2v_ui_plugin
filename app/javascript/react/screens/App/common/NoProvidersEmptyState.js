import React from 'react';
import PropTypes from 'prop-types';
import ShowWizardEmptyState from './ShowWizardEmptyState/ShowWizardEmptyState';

const NoProvidersEmptyState = ({ className }) => (
  <ShowWizardEmptyState
    className={className}
    description={
      __('Before attempting a migration, you must have at least one VMware and one Red Hat Virtualization or Red Hat OpenStack Platform provider configured.') // prettier-ignore
    }
    action={
      <React.Fragment>
        <a href="/ems_infra/show_list">{__('View VMware and Red Hat Virtualization providers')}</a>
        <br />
        <a href="/ems_cloud/show_list">{__('View Red Hat OpenStack Platform providers')}</a>
      </React.Fragment>
    }
  />
);

NoProvidersEmptyState.propTypes = {
  className: PropTypes.string
};

export default NoProvidersEmptyState;
