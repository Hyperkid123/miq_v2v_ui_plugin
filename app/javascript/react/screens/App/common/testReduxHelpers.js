import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';

const middlewares = [thunk, promiseMiddleware()];

export const generateStore = (reducers, initialState) =>
  createStore(combineReducers(reducers), initialState, applyMiddleware(...middlewares));
