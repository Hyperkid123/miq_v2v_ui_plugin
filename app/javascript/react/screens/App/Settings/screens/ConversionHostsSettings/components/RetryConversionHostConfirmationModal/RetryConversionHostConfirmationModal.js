import React from 'react';
import PropTypes from 'prop-types';
import { reduxForm, Field } from 'redux-form';
import { required } from 'redux-form-validators';
import { Button, Modal, Form, Grid } from 'patternfly-react';
import TextFileInput from '../../../../../common/forms/TextFileInput';
import { FormField } from '../../../../../common/forms/FormField';
import { V2V_TARGET_PROVIDERS, OPENSTACK, RHV } from '../../../../../../../../common/constants';

const requiredWithMessage = required({ msg: __('This field is required') });
const bodyIsRequired = value => requiredWithMessage(value.body);
const fieldBaseProps = { labelWidth: 4, controlWidth: 7 };

const RetryConversionHostConfirmationModal = ({
  show,
  conversionHostTaskToRetry,
  isPostingConversionHosts,
  hideConversionHostRetryModalAction,
  conversionHostRetryModalExitedAction,
  postConversionHostsAction,
  postConversionHostsUrl,
  retryForm
}) => {
  const requestParams =
    conversionHostTaskToRetry &&
    conversionHostTaskToRetry.context_data &&
    conversionHostTaskToRetry.context_data.request_params;
  const isUsingSshTransformation = !requestParams.vmware_vddk_package_url;
  const selectedProviderType = V2V_TARGET_PROVIDERS.find(provider => provider.type === requestParams.resource_type).id;
  // TODO dedupe this from the auth step of the wizard:
  let sshKeyInfo = '';
  if (selectedProviderType === RHV) {
    sshKeyInfo = __('RHV-M deploys a common SSH public key on all hosts when configuring them. This allows commands and playbooks to be run from RHV-M. The associated private key is in the file /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M.'); // prettier-ignore
  }
  if (selectedProviderType === OPENSTACK) {
    sshKeyInfo = __('This is the private key file used to connect to the conversion host instance for the OpenStack User.'); // prettier-ignore
  }

  const formHasErrors = retryForm && !!retryForm.syncErrors;

  return (
    <Modal
      show={show}
      onHide={hideConversionHostRetryModalAction}
      onExited={conversionHostRetryModalExitedAction}
      dialogClassName="modal-lg wizard-pf"
      backdrop="static"
    >
      <Modal.Header>
        <Modal.CloseButton onClick={hideConversionHostRetryModalAction} />
        <Modal.Title>{__('Retry Conversion Host Configuration')}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Form className="form-horizontal">
          <ul>
            <p>
              {__('For your security, private SSH keys from the configuration wizard were not saved.')} <br />
              {__('Re-enter the necessary authentication details to retry configuration of the following conversion host:') /* prettier-ignore */}
            </p>
          </ul>
          <Form.FormGroup>
            <Grid.Col componentClass={Form.ControlLabel} sm={fieldBaseProps.labelWidth}>
              {__('Host')}
            </Grid.Col>
            <Grid.Col sm={fieldBaseProps.controlWidth} style={{ paddingTop: 3 }}>
              {conversionHostTaskToRetry.name}
            </Grid.Col>
          </Form.FormGroup>
          <Field
            {...fieldBaseProps}
            name="conversionHostSshKey"
            label={__('Conversion Host SSH private key')}
            component={FormField}
            info={sshKeyInfo}
            controlId="host-ssh-key-input"
            required
            validate={[bodyIsRequired]}
          >
            {({ input: { value, onChange, onBlur } }) => (
              <TextFileInput
                help={__('Upload your SSH key file or paste its contents below.')}
                value={value}
                onChange={onChange}
                onBlur={onBlur}
              />
            )}
          </Field>
          {isUsingSshTransformation && (
            <Field
              {...fieldBaseProps}
              name="vmwareSshKey"
              label={__('VMware hypervisors SSH private key')}
              component={FormField}
              controlId="vmware-ssh-key-input"
              required
              validate={[bodyIsRequired]}
            >
              {({ input: { value, onChange, onBlur } }) => (
                <TextFileInput
                  help={__('Upload your SSH key file or paste its contents below.')}
                  value={value}
                  onChange={onChange}
                  onBlur={onBlur}
                />
              )}
            </Field>
          )}
        </Form>
      </Modal.Body>
      <Modal.Footer>
        <Button bsStyle="default" className="btn-cancel" onClick={hideConversionHostRetryModalAction}>
          {__('Cancel')}
        </Button>
        <Button
          bsStyle="primary"
          disabled={formHasErrors || isPostingConversionHosts}
          onClick={() => {
            const postBody = { ...conversionHostTaskToRetry.context_data.request_params };
            // TODO also pass the two SSH keys
            postConversionHostsAction(postConversionHostsUrl, [postBody]);
          }}
        >
          {__('Retry')}
        </Button>
      </Modal.Footer>
    </Modal>
  );
};

RetryConversionHostConfirmationModal.propTypes = {
  show: PropTypes.bool,
  conversionHostTaskToRetry: PropTypes.object,
  isPostingConversionHosts: PropTypes.bool,
  hideConversionHostRetryModalAction: PropTypes.func,
  conversionHostRetryModalExitedAction: PropTypes.func,
  postConversionHostsAction: PropTypes.func,
  postConversionHostsUrl: PropTypes.string,
  retryForm: PropTypes.object
};

export default reduxForm({
  form: 'retryConversionHost',
  initialValues: {
    conversionHostSshKey: { filename: '', body: '' },
    vmwareSshKey: { filename: '', body: '' }
  }
})(RetryConversionHostConfirmationModal);

// TODO add the keys to the request
// TODO prevent Retry button in the list view from appearing if there is no context_data.request_params
// TODO dedupe the sshKeyInfo, and maybe even the actual fields, abstracted along with the auth step of the wizard
// TODO refactor the whole list view to be in its own folder with an index.js
