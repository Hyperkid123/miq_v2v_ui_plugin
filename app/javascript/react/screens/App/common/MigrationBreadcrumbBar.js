import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumb } from 'patternfly-react';
import Toolbar from '../../../config/Toolbar';
import BreadcrumbPageSwitcher from './BreadcrumbPageSwitcher';

const items = [
  { name: __('Migration Plans'), href: '#/plans' },
  { name: __('Infrastructure Mappings'), href: '#/mappings' },
  { name: __('Migration Settings'), href: '#/settings' }
];

const MigrationBreadcrumbBar = ({ activeHref }) => {
  const activeItem = items.find(item => item.href === activeHref);
  return (
    <Toolbar>
      <Breadcrumb.Item active>{__('Migration')}</Breadcrumb.Item>
      <Breadcrumb.Item active>
        <strong>{activeItem.name}</strong>
      </Breadcrumb.Item>
      <BreadcrumbPageSwitcher items={items} activeHref={activeHref} />
    </Toolbar>
  );
};

MigrationBreadcrumbBar.propTypes = {
  activeHref: PropTypes.string.isRequired
};

export default MigrationBreadcrumbBar;
