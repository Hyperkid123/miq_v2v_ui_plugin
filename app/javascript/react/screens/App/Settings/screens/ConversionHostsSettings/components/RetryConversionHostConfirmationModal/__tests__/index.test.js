import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';

import { generateStore } from '../../../../../../common/testReduxHelpers';
import settingsReducer, { initialState as settingsInitialState } from '../../../../../SettingsReducer';
import RetryConversionHostConfirmationModal from '../RetryConversionHostConfirmationModal';
import RetryConversionHostConfirmationModalContainer from '../index';

describe('ConversionHostWizardLocationStep integration test', () => {
  const store = generateStore(
    { form: formReducer, settings: settingsReducer },
    {
      form: { retryConversionHost: {} },
      settings: {
        ...settingsInitialState,
        conversionHostTaskToRetry: {
          context_data: { request_params: { mock: 'params', vmware_vddk_package_url: 'mock vddk path' } }
        }
      }
    }
  );

  const mountComponent = () =>
    mount(
      <Provider store={store}>
        <RetryConversionHostConfirmationModalContainer />
      </Provider>
    );

  it('should mount ConversionHostWizardLocationStep with mapStateToProps reduced', () => {
    const wrapper = mountComponent();

    // query the unconnected component to assert reduced props
    const component = wrapper.find(RetryConversionHostConfirmationModal);
    expect(component.props()).toMatchSnapshot();
  });
});
