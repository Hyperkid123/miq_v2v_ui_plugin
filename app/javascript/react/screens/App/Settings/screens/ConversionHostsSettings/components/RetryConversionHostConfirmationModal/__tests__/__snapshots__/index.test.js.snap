// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversionHostWizardLocationStep integration test should mount ConversionHostWizardLocationStep with mapStateToProps reduced 1`] = `
Object {
  "conversionHostRetryModalExitedAction": [Function],
  "conversionHostTaskToRetry": Object {
    "context_data": Object {
      "request_params": Object {
        "mock": "params",
        "vmware_vddk_package_url": "mock vddk path",
      },
    },
  },
  "hideConversionHostRetryModalAction": [Function],
  "isPostingConversionHosts": false,
  "postConversionHostsAction": [Function],
  "retryForm": Object {
    "initial": Object {
      "conversionHostSshKey": Object {
        "body": "",
        "filename": "",
      },
      "vmwareSshKey": Object {
        "body": "",
        "filename": "",
      },
    },
    "values": Object {
      "conversionHostSshKey": Object {
        "body": "",
        "filename": "",
      },
      "vmwareSshKey": Object {
        "body": "",
        "filename": "",
      },
    },
  },
  "show": false,
}
`;
